### ISSUE TITLE
Deprecated notice (events)

### ISSUE THREAD

**Issue by thepurpleblob** [ULCC/open-mod_coursework#2]
Events API using $handlers array has been deprecated in favour of Events 2 API, please use it instead.
line 4050 of /lib/deprecatedlib.php: call to debugging()
line 710 of /lib/upgradelib.php: call to events_update_definition()
line 432 of /lib/upgradelib.php: call to upgrade_plugins_modules()
line 1740 of /lib/upgradelib.php: call to upgrade_plugins()
line 677 of /admin/index.php: call to upgrade_noncore()

**Comment by winko7** [comment]
Events API using $handlers array has been deprecated in favour of Events 2 API, please use it instead.

```
line 3938 of \lib\deprecatedlib.php: call to debugging()
line 4216 of \lib\deprecatedlib.php: call to events_dispatch()
line 809 of \lib\classes\event\base.php: call to events_trigger_legacy()
line 169 of \course\modlib.php: call to core\event\base->trigger()
line 275 of \course\modedit.php: call to add_moduleinfo()
```

Fatal error: Can't use function return value in write context in \server\moodle\mod\coursework\classes\stages\base.php on line 232

**Comment by aspark21** [comment]
This was fixed in v16.09.23 / 16be184
